bolts/data/datasets/utils.py

```diff
@@ -108,6 +112,23 @@ def img_to_tensor(img: Image.Image | np.ndarray) -> Tensor:
     )
 
 
+AudioLoadingBackend = Literal["sox_io", "soundfile"]
+
+AudioTform = Union[TF_audio.Spectrogram, TF_audio.MelSpectrogram]
+
+
+def infer_al_backend() -> AudioLoadingBackend:
+    """Infer which audio-loading backend to use based on the operating system."""
+    return 'soundfile' if platform.system() == 'Windows' else 'sox_io'
+
+
+def apply_waveform_transform(waveform: Tensor, *, transform: AudioTform | None) -> Tensor:
+    waveform_ = waveform
+    if transform is not None:
+        waveform_ = transform(waveform_)
+    return waveform_
```

Comment:
although there's not a hard rule about it, usually the trailing `_` is for when you want to use a python reserved word, like `class_`. But in this case, I think it's not needed.
```suggestion
   return waveform if transform is None else transform(waveform)
```

Reply:
Ahh ok, I know there's a lot of unspoken rules in python, good to learn some more.